lumicks/pylake/population/hmm.py

```diff
@@ -0,0 +1,210 @@
+import numpy as np
+
+from ..channel import Slice
+from .dwelltime import _dwellcounts_from_statepath
+from .detail.hmm import ClassicHmm, viterbi, baum_welch
+from .detail.validators import col
+
+
+class HiddenMarkovModel:
+    """A Hidden Markov Model describing hidden state occupancy and state transitions of observed
+    time series data (force, fluorescence, etc.)
+
+    A detailed description of the model properties and training algorithms can be found in [1]_.
+
+    .. warning::
+
+        This is early access alpha functionality. While usable, this has not yet been tested in a
+        large number of different scenarios. The API can still be subject to change without any
+        prior deprecation notice! If you use this functionality keep a close eye on the changelog
+        for any changes that may affect your analysis.
+
+    Parameters
+    ----------
+    data : numpy.ndarray | Slice
+        Data array used for model training.
+    n_states : int
+        The number of hidden states in the model.
+    tol : float
+        The tolerance for training convergence.
+    max_iter : int
+        The maximum number of iterations to perform.
+    gmm_guess : GaussianMixtureModel
+        Initial guess for the observation model parameters.
```

Comment:
Maybe not for this PR (it's big enough 😅) but it would be nice if we could pass in an hmm as well. If a previous fit didn't converge and it took a while, we could continue the run?